### PR TITLE
Extract public module variables from ColonizationAI

### PR DESCRIPTION
**Pull request by Cjkjvfnby**
This is refactoring.  I don't expect any output. 

The interesting part for review could be the README.md file.  It describes my ideas behind this refactoring.

### CODE REVIEW

File: default/python/AI/empire/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Empire info module
+
+This module contains code that responsible for holding useful information about empire objects.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `code responsible` or `code that is resonsible`

File: default/python/AI/empire/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Implementation details. 
+
+Files in this module wildly use encapsulation principle.
+
+Data container itself is hidden from outer scope and access to it should be provided via functions.
+This approach allows changing internal implementation and add new functionality
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `and adding` or `to add`

File: default/python/AI/empire/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+This module contains code that responsible for holding useful information about empire objects.
+
+## Implementation details. 
+
+Files in this module wildly use encapsulation principle.
+
+Data container itself is hidden from outer scope and access to it should be provided via functions.
+This approach allows changing internal implementation and add new functionality 
+without changing code which uses this module.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `which` -> `that`

File: default/python/AI/empire/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+This module contains code that responsible for holding useful information about empire objects.
+
+## Implementation details. 
+
+Files in this module wildly use encapsulation principle.
+
+Data container itself is hidden from outer scope and access to it should be provided via functions.
+This approach allows changing internal implementation and add new functionality 
+without changing code which uses this module. 
+Also, we could keуp this module as simple as required for current code.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `keyp` -> `keep`

Not sure what this sentence means, though.

File: default/python/AI/empire/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+Modules usually don't have any AI dependencies, so they could be easily used from any part of AI code.
+
+Each module have fucntion to set data and functions to get data. This gives us a temporal coupling.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `have fucntion` -> `has functions` or `has a function` (not sure how many functions are present)

File: default/python/AI/empire/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+Data container itself is hidden from outer scope and access to it should be provided via functions.
+This approach allows changing internal implementation and add new functionality 
+without changing code which uses this module. 
+Also, we could keуp this module as simple as required for current code.
+
+Modules usually don't have any AI dependencies, so they could be easily used from any part of AI code.
+
+Each module have fucntion to set data and functions to get data. This gives us a temporal coupling. 
+Using any getter before all setters calls will lead to unexpected behaviour. Be aware of it.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `setters` -> `setter`

File: default/python/AI/empire/README.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+This approach allows changing internal implementation and add new functionality 
+without changing code which uses this module. 
+Also, we could keуp this module as simple as required for current code.
+
+Modules usually don't have any AI dependencies, so they could be easily used from any part of AI code.
+
+Each module have fucntion to set data and functions to get data. This gives us a temporal coupling. 
+Using any getter before all setters calls will lead to unexpected behaviour. Be aware of it. 
+
+To achieve encapsulation the following is done:
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `,` after `encapsulation`

File: default/python/AI/empire/README.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+The main data structure is wrapped to the `@cache_for_current_turn`. 
+So basically you will get new value only once per turn.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `,` after `basically`

Comment (geoffthemedio) [reply default/python/AI/empire/README.md]: `will get new value` -> `can get a new value` or `will get a new value`

File: default/python/AI/empire/README.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+Modules usually don't have any AI dependencies, so they could be easily used from any part of AI code.
+
+Each module have fucntion to set data and functions to get data. This gives us a temporal coupling. 
+Using any getter before all setters calls will lead to unexpected behaviour. Be aware of it. 
+
+To achieve encapsulation the following is done:
+The main data structure is wrapped to the `@cache_for_current_turn`. 
+So basically you will get new value only once per turn. 
+This approach require a few lines of code it is hard to forget to make a cleanup before each turn.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `require` -> `requires`

Seems like a run on sentence...  not sure what is meant. Maybe `.` after `code` ?

File: default/python/AI/empire/README.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+This might be a problem when writing unittests. Probably will need new decorator for this.
```
Comment (geoffthemedio) [inline default/python/AI/empire/README.md]: `need new` -> `need a new` ?

**Comment by geoffthemedio** [comment]
@Cjkjvfnby any idea what's happening with the Python code formatting check? It's also failing for #3541

File: default/python/AI/empire/README.md (hunk 4 above)
Comment (Cjkjvfnby) [reply default/python/AI/empire/README.md]: I try to say that we could implement the only API required for current AI code and later add more methods and change the inner structure to support them. 

For example, I have changed `colonies_under_attack` from a list of colonies to bool. It is easy to change it back when we will need it.  And at the current state, this is the simplest possible implementation.

**Comment by Cjkjvfnby** [comment]
@geoffthemedio, thank you for the review. 

> any idea what's happening with the Python code formatting check?

Not really. This is a bit strange. 

Important info is missed in `isort` output, a fix will look like: (https://github.com/freeorion/freeorion/pull/3531/files#diff-d7578ac1d58c70927a7a00a5bc7c8f3934a4e4ec5643021c9d3081e731af7be9R28)
I
 have managed to beat it in PR with black, but not sure how.

**Comment by Cjkjvfnby** [comment]
@geoffthemedio 

There is a feature named suggestion in GitHub [see item 6](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request). 

When I do a review it is a bit more convenient for me to use it for small fixes. It requires fewer actions.

**Pull request merged**